docs/src/pages/style/color/ColorTool.js

```diff
@@ -0,0 +1,282 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import compose from 'recompose/compose';
+import { withStyles } from '@material-ui/core/styles';
+import Grid from '@material-ui/core/Grid';
+import Input from '@material-ui/core/Input';
+import Radio from '@material-ui/core/Radio';
+import Tooltip from '@material-ui/core/Tooltip';
+import Typography from '@material-ui/core/Typography';
+import Button from '@material-ui/core/Button';
+import CheckIcon from '@material-ui/icons/Check';
+import * as colors from '@material-ui/core/colors';
+import Slider from '@material-ui/lab/Slider';
+import actionTypes from 'docs/src/modules/redux/actionTypes';
+import ColorDemo from './ColorDemo';
+
+const hues = [
+  'red',
+  'pink',
+  'purple',
+  'deepPurple',
+  'indigo',
+  'blue',
+  'lightBlue',
+  'cyan',
+  'teal',
+  'green',
+  'lightGreen',
+  'lime',
+  'yellow',
+  'amber',
+  'orange',
+  'deepOrange',
+];
+
+const shades = [900, 800, 700, 600, 500, 400, 300, 200, 100, 50, 'A700', 'A400', 'A200', 'A100'];
+
+export const styles = theme => ({
+  root: {
+    display: 'flex',
+    width: '100%',
+  },
+  radio: {
+    width: 47,
+    height: 47,
+  },
+  radioSelected: {
+    width: 47,
+    height: 47,
+    border: '1px solid white',
+    color: theme.palette.common.white,
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  sliderContainer: {
+    display: 'flex',
+    justifyContent: 'spece-between',
+    width: 192,
+    marginTop: theme.spacing.unit * 2,
+    marginBottom: theme.spacing.unit,
+  },
+  sliderTypography: {
+    marginTop: 5,
+  },
+  swatch: {
+    width: 192,
+    backgroundColor: theme.palette.common.white,
+  },
+  input: {
+    marginTop: 24,
+    width: 192,
+  },
+  button: {
+    marginTop: 24,
+  },
+});
+
+class ColorTool extends React.Component {
+  state = {
+    primary: '#2196f3',
+    secondary: '#f50057',
+    primaryText: '#2196f3',
+    secondaryText: '#f50057',
+    primaryHue: 'blue',
+    secondaryHue: 'pink',
+    primaryShade: 4,
+    secondaryShade: 11,
+  };
+
+  hashPrefix = string => (string.charAt(0) === '#' ? string : `#${string}`);
+
+  isRgb = string => /#?([0-9a-f]{6})/i.test(string);
```

Comment:
Why using a class method over a function on the root scope?

Reply:
Do you have a rule for when to use which?

Reply:
class methods can access the instance. I try to promote the function that doesn't need it to the top level to increase the isolation between the different sections of the code.